msticpy/data/drivers/local_velociraptor_driver.py

```diff
@@ -0,0 +1,222 @@
+
+        Parameters
+        ----------
+        connection_str : str, optional
+            Connection string (not used)
+        data_paths : List[str], optional
+            Paths from which to load data files
+        progress : bool, optional
+            Show progress with tqdm, by default, True
+
+        """
+        del connection_str
+        if kwargs.get("debug", False):
+            logger.setLevel(logging.DEBUG)
+        super().__init__()
+
+        self._paths: List[str] = ["."]
+        # If data paths specified, use these
+        # from kwargs or settings
+        if data_paths := kwargs.get("data_paths"):
+            self._paths = [path.strip() for path in data_paths]
+            logger.info("data paths read from param %s", str(self._paths))
+        else:
+            prov_settings = get_provider_settings(config_section="DataProviders").get(
+                "VelociraptorLogs"
+            )
+            if prov_settings:
+                self._paths = prov_settings.args.get("data_paths", []) or self._paths
+                logger.info("data paths read from config %s", str(self._paths))
+
+        self.data_files: Dict[str, List[Path]] = {}
+        self._schema: Dict[str, Any] = {}
+        self._query_map: Dict[str, List[str]]
+        self._progress = kwargs.pop("progress", True)
+        self._loaded = True
+        self.has_driver_queries = True
+        logger.info("data files to read %s", ",".join(self.data_files))
+
+    def connect(self, connection_str: Optional[str] = None, **kwargs):
+        """
+        Connect to data source.
+
+        Parameters
+        ----------
+        connection_str : str
+            Connect to a data source
+
+        """
+        del connection_str
+        self.data_files = self._get_logfile_paths()
+        self._connected = True
+
+    @property
+    def schema(self) -> Dict[str, Dict]:
+        """
+        Return current data schema of connection.
+
+        Returns
+        -------
+        Dict[str, Dict]
+            Data schema of current connection.
+
+        """
+        if not self._schema:
+            if not self.data_files:
+                self.connect()
+            # read the first row of each file to get the schema
+            iter_data_files = (
+                tqdm(self.data_files.items())
+                if self._progress
+                else self.data_files.items()
+            )
+            for table, files in iter_data_files:
+                if not files:
+                    continue
+                sample_df = pd.read_json(files[0], lines=True, nrows=1)
+                self._schema[table] = {
+                    col: dtype.name for col, dtype in sample_df.dtypes.to_dict().items()
+                }
+            logger.info("Reading schema for %d tables", len(self.data_files))
+
+        return self._schema
+
+    def query(
+        self, query: str, query_source: Optional[QuerySource] = None, **kwargs
+    ) -> Union[pd.DataFrame, Any]:
```

Comment:
Should the return type annotation be `Any` or `results.ResultSet`?

Reply:
The base class annotations are pd.DataFrame, Any. I sometimes violate this in subclasses where there is something more specific that's reasonable but it isn't ideal.